src/sentry/sentry_metrics/consumers/indexer/processing.py

```diff
@@ -53,15 +60,18 @@ def process_messages(
     else:
         # This, instead of importing the normal way, was likely done to prevent
         # fork-safety issues.
-        indexer = get_indexer()
+        if config.db_backend == "cloudspanner":
+            indexer = get_spanner_indexer()
+        else:
+            indexer = get_indexer()
```

Comment:
How will sentry pick the right indexer to reverse resolve ? 
For this test we do not need to care, but we will have to cut over at a point. Do you envision us doing it through multiple implementations of the service delegator ?

Reply:
I think eventually we could load the cloudspanner be loaded via the lazy service wrapper, and ideally `from sentry.sentry_metrics import indexer` can just work. Right now since I'm adding a cli arg, that doesn't seem to mesh well with loading from settings. It also seems to me like if we want to be easily able to cut over or switch between the two, we'd wanna stay with cli arg to be make those changes more quickly. anyway just my initial thoughts, would have to think a bit more deeply on it

Reply:
There are currently two entrypoints to the indexer - one via the product/API and one via ingest. These use cases are quite different but we have to force the interface of the indexer to be the same because of this. Since we are running multiple indexers for ingest I don't think it makes any sense to use the service delegator there as service delegation is mostly useful when you want your users to be able to switch between backends easily. I would propose keeping the service delegator but making it only for Sentry/API use cases and not on the ingest path. In this scenario the in this method the Postgres indexer would just be loaded directly (like the cloudspanner one is) and not via service delegator. What do you think?

Reply:
Updated to have the `process_messages` function choose the indexer via the `STORAGE_TO_INDEXER` mapping instead of the `LazyServiceWrapper`